flink-core/src/main/java/org/apache/flink/api/common/functions/RuntimeContext.java

```diff
@@ -63,14 +67,27 @@ public interface RuntimeContext {
     /**
      * The ID of the current job. Note that Job ID can change in particular upon manual restart. The
      * returned ID should NOT be used for any job management tasks.
+     *
+     * @deprecated This method is deprecated since Flink 1.19. All metadata about the job should be
+     *     provided uniformly by {@link #getJobInfo()}.
+     * @see <a
+     *     href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-382%3A+Unify+the+Provision+of+Diverse+Metadata+for+Context-like+APIs">
+     *     FLIP-382: Unify the Provision of Diverse Metadata for Context-like APIs </a>
      */
+    @Deprecated
     JobID getJobId();
```

Comment:
Actually, we can provide default implementations for these deprecated methods. E.g., `JobID getJobId() { return getJobInfo().getJobId(); }`, so that we can remove the duplicated method implementations for the concrete classes.

Reply:
Good point! For these deprecated methods in the modified interfaces, I've changed them from abstract to default and provide default implementations.